tests/compiler/std/set.ts

```diff
@@ -40,62 +42,88 @@ function testNumeric<K extends number>(): void {
   assert(set.size == 0);
 }
 
-test<i8>();
-test<u8>();
-test<i16>();
-test<u16>();
-test<i32>();
-test<u32>();
-test<i64>();
-test<u64>();
-test<f32>();
-test<f64>();
-
-// forEach
-var set = new Set<i32>();
-for (let k: i32 = 0; k < 10; ++k) {
-  set.add(k);
-}
-set.add(124);
-set.delete(124);
-assert(set.size == 10);
-var i: i32 = 0;
-set.forEach((value1: i32, value2: i32, aset: Set<i32>): void => {
-  assert(value1 == value2);
-  assert(i == value1);
-  i++;
-});
-assert(i == 10);
-
-i = 0;
-set.forEach((value1: i32, value2: i32, aset: Set<i32>): void => {
-  aset.add(11);
-  i++;
-});
-assert(set.size == 11);
-assert(11 == i);
-
-i = 0;
-set.forEach((value1: i32, value2: i32, aset: Set<i32>): void => {
-  if (value1 == 11) {
-    for (let index = 0; index < 100; index++) {
-      aset.add(index + 20);
-    }
+testNumeric<i8>();
+testNumeric<u8>();
+testNumeric<i16>();
+testNumeric<u16>();
+testNumeric<i32>();
+testNumeric<u32>();
+testNumeric<i64>();
+testNumeric<u64>();
+testNumeric<f32>();
+testNumeric<f64>();
+
+var i = 0;
+function testForEach(): void {
+  var set = new Set<i32>();
+  for (let k: i32 = 0; k < 10; ++k) {
+    set.add(k);
   }
-  i++;
-});
-assert(set.size == 111);
-assert(set.size == i);
-
-i = 0;
-set.forEach((value1: i32, value2: i32, aset: Set<i32>): void => {
-  aset.delete(value1);
-  i++;
-});
-assert(set.size == 0);
-assert(i == 111);
-
-i = 0;
-set.forEach((value1: i32, value2: i32, aset: Set<i32>): void => {
-  i++;
-});
+  assert(set.size == 10);
+
+  set.forEach((value1: i32, value2: i32, aset: Set<i32>): void => {
```

Comment:
You could use `value1 => { ... }` currently. Or if lang server/lint not happy adding `(value1: i32): void => { ... }`

Reply:
That's good.

Reply:
And for rest tests as well if you don't mind